BookingsApi/BookingsApi.IntegrationTests/Api/V2/Hearings/UpdateHearingsInGroupV2Tests.cs

```diff
@@ -569,88 +571,100 @@ private void AssertEventsPublished(Hearing hearing, HearingRequestV2 requestHear
                     .Select(x => x.IntegrationEvent as EndpointAddedIntegrationEvent)
                     .Where(x => x.HearingId == hearing.Id)
                     .ToList();
-
                 endpointAddedMessages.Count.Should().Be(expectedAddedCount);
-                
                 var expectedUpdatedCount = requestHearing.Endpoints.ExistingEndpoints.Count;
-
                 var endpointUpdatedMessages = messages
                     .Where(x => x.IntegrationEvent is EndpointUpdatedIntegrationEvent)
                     .Select(x => x.IntegrationEvent as EndpointUpdatedIntegrationEvent)
                     .Where(x => x.HearingId == hearing.Id)
                     .ToList();
-
                 endpointUpdatedMessages.Count.Should().Be(expectedUpdatedCount);
-                
                 var expectedRemovedCount = requestHearing.Endpoints.RemovedEndpointIds.Count;
-                
                 var endpointRemovedMessages = messages
                     .Where(x => x.IntegrationEvent is EndpointRemovedIntegrationEvent)
                     .Select(x => x.IntegrationEvent as EndpointRemovedIntegrationEvent)
                     .Where(x => x.HearingId == hearing.Id)
                     .ToList();
-
                 endpointRemovedMessages.Count.Should().Be(expectedRemovedCount);
             }
         }
+        
+        private void AssertNotificationEvents(Hearing hearing, HearingRequestV2 requestHearing, int existingParticipantsUpdated)
+        {
+            var serviceBusStub = Application.Services
+                .GetService(typeof(IServiceBusQueueClient)) as ServiceBusQueueClientFake;
+            var messages = serviceBusStub!
+                .ReadAllMessagesFromQueue(hearing.Id);
+            
+            
+            var existingParticipantMessages = messages
+                .Where(x => x.IntegrationEvent is ExistingParticipantMultidayHearingConfirmationEvent)
+                .Select(x => x.IntegrationEvent as ExistingParticipantMultidayHearingConfirmationEvent)
+                .Where(x => x.HearingConfirmationForParticipant.HearingId == hearing.Id)
+                .ToList();
+            
+            var newParticipantMessages = messages
+                .Where(x => x.IntegrationEvent is NewParticipantMultidayHearingConfirmationEvent)
+                .Select(x => x.IntegrationEvent as NewParticipantMultidayHearingConfirmationEvent)
+                .Where(x => x.HearingConfirmationForParticipant.HearingId == hearing.Id)
+                .ToList();
+            
+            var newParticipantWelcomeMessages = messages
+                .Where(x => x.IntegrationEvent is NewParticipantWelcomeEmailEvent)
+                .Select(x => x.IntegrationEvent as NewParticipantWelcomeEmailEvent)
+                .Where(x => x.WelcomeEmail.HearingId == hearing.Id)
+                .ToList();
+            
+            existingParticipantMessages.Count.Should().Be(3);
```

Comment:
these hardcoded counts make this method difficult to re-use. Could we either calculate the counts from the passing in hearing request, or pass in the expected counts as arguments (eg `private void AssertNotificationEvents(int expectedParticipantMessageCount, int newParticipantMessageCount`...)

Reply:
done